### PR TITLE
created middleware for generating error pages

### PR DESCRIPTION
**Pull request by jeromes123**
**OVERVIEW**
1. created middleware and implemented to handle access of unauthenticated and authenticated users.

### CODE REVIEW

File: src/routes/web.php [hunk 1]
```diff
@@ -35,37 +24,35 @@
...
     Route::get('/edit-post-page/{post}', [UserPostController::class, 'showEditPostPage']);
     Route::put('/edit-post', [UserPostController::class, 'edit']);
     Route::delete('/delete-post/{post}', [UserPostController::class, 'delete']);
     Route::post('/like-post/{post}', [UserPostController::class, 'likePost']);
     Route::delete('/unlike-post/{post}', [UserPostController::class, 'unlikePost']);
+    Route::get('/logout', [SigninController::class, 'logout']);
+    Route::post('/update-profile', [ProfileController::class, 'update']);
+
+    Route::get('/update-profile', function () {
+        $profile = Profile::where('user_id', auth()->user()->id)->firstOrFail();
```
Comment (mikco-hipe) [inline src/routes/web.php]: this should not be here

Comment (jeromes123) [reply src/routes/web.php]: done

File: src/app/Http/Controllers/ProfileController.php [hunk 2]
```diff
@@ -27,4 +28,14 @@ public function update(UpdateProfileRequest $request): RedirectResponse
...
         return redirect('/update-profile'); 
     }    
+
+    /**
+     * Displays the page to update profile information.
+     */
+    public function showUpdateProfile(): View
+    {
+        return view('components.create-profile', [
+            'profile' => Profile::where('user_id', auth()->user()->id)->firstOrFail()
```
Comment (mikco-hipe) [inline src/app/Http/Controllers/ProfileController.php]: Transfer this service

Comment (jeromes123) [reply src/app/Http/Controllers/ProfileController.php]: will do

**Comment by jeromes123** [comment]
Preview of error pages

image
image
image

**Review by mikco-hipe — APPROVED** [review]
LGTM 🙇

**Review by mikco-hipe — APPROVED** [review]

**Comment by jeromes123** [comment]
resolved conflicts on routes

**Pull request merged**